apps/rabbitmq_prelaunch/src/rabbit_prelaunch.erl

```diff
@@ -91,12 +91,16 @@ do_run() ->
 
     %% Complete context now that we have the final environment loaded.
     Context2 = rabbit_env:get_context_after_reloading_env(Context1),
-    ?assertMatch(#{}, Context2),
-    store_context(Context2),
-    rabbit_env:log_context(Context2),
+
+    %% Migrate the enabled_plugins file to the new location if necessary
+    Context3 = rabbit_plugins:maybe_migrate_enabled_plugins_file(Context2),
```

Comment:
The `rabbitmq_prelaunch` application can't depend on a module from `rabbit`. The dependency is the other way around.

You could move the code you added to `rabbit_plugins` to a new `rabbitmq_prelaunch_plugins` module in the `rabbitmq_prelaunch` application for instance.

Reply:
Ok, that makes sense.

I see we don't appear to separate the prelaunch app's tests into their own directory? So I will leave the `rabbit_plugins_SUITE` in `test/` as `rabbit_prelaunch_plugins_SUITE` instead of placing it in `apps/rabbit_prelaunch/test`.